File: src/main/java/mods/eln/sim/ThermalLoadInitializer.java
package mods.eln.sim;

import mods.eln.Eln;
import mods.eln.sim.process.destruct.ThermalLoadWatchDog;

public class ThermalLoadInitializer {

    public double maximumTemperature, minimumTemperature;
    double heatingTao;
    double conductionTao;

    double Rs, Rp, C;

    public ThermalLoadInitializer(double maximumTemperature, double minimumTemperature, double heatingTao, double conductionTao) {
        this.conductionTao = conductionTao;
        this.minimumTemperature = minimumTemperature;
        this.heatingTao = heatingTao;
        this.maximumTemperature = maximumTemperature;
    }

    public void setMaximalPower(double power) {
        C = power * heatingTao / (maximumTemperature);
        Rp = maximumTemperature / power;
        Rs = conductionTao / C / 2;

        Eln.simulator.checkThermalLoad(Rs, Rp, C);
    }

    public void applyTo(ThermalLoad load) {
        load.set(Rs, Rp, C);
    }

    public void applyTo(ThermalLoadWatchDog doggy) {
        doggy.setThermalLoad(this);
    }

    public ThermalLoadInitializer copy() {
        ThermalLoadInitializer thermalLoad = new ThermalLoadInitializer(maximumTemperature, minimumTemperature, heatingTao, conductionTao);
        thermalLoad.Rp = Rp;
        thermalLoad.Rs = Rs;
        thermalLoad.C = C;
        return thermalLoad;
    }
}

File: src/main/java/mods/eln/sim/thermal/ThermalLoadInitializer.kt
package mods.eln.sim.thermal

class ThermalLoadInitializer(var warmLimit: Double, var coolLimit: Double, internal var heatingTao: Double, internal var conductionTao: Double) {

    internal var Rs: Double = 0.toDouble()
    internal var Rp: Double = 0.toDouble()
    internal var C: Double = 0.toDouble()

    /*	public ThermalLoadInitializer (
            double warmLimit,double coolLimit,
			double heatingTao,double conductionTao,
			double P) {
		this.conductionTao = conductionTao;
		this.coolLimit = coolLimit;
		this.heatingTao = heatingTao;
		this.warmLimit = warmLimit;
		setMaximalPower(P);
	}*/

    fun setMaximalPower(P: Double) {
        C = P * heatingTao / warmLimit
        Rp = warmLimit / P
        Rs = conductionTao / C / 2.0

        ThermalLoad.checkThermalLoad(Rs, Rp, C)
    }

    fun applyTo(load: ThermalLoad) {
        load.set(Rs, Rp, C)
    }

    fun applyTo(doggy: ThermalLoadWatchDog) {
        doggy.set(this)
    }

    fun copy(): ThermalLoadInitializer {
        val thermalLoad = ThermalLoadInitializer(warmLimit, coolLimit, heatingTao, conductionTao)
        thermalLoad.Rp = Rp
        thermalLoad.Rs = Rs
        thermalLoad.C = C
        return thermalLoad
    }
}
